presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -1449,6 +1457,30 @@ public void verifyComparableOrderableContract()
         }
     }
 
+    private static boolean hasEqualsMethod(Type type)
+    {
+        try {
+            OperatorInvoker equalsOperator = type.getEqualsOperator(simpleConvention(NULLABLE_RETURN, NEVER_NULL, NEVER_NULL));
```

Comment:
What if there was an equals operator with a different convention? (Unless we don't yet support that, in which case, add a comment or TODO)